pxtlib/docsrender.ts

```diff
@@ -437,16 +437,16 @@ namespace pxt.docs {
             if (m) {
                 text = m[1]
                 id = m[2]
-            } else {
-                id = raw.toLowerCase().replace(/[^\w]+/g, '-')
             }
             // remove tutorial macros
             if (text)
                 text = text.replace(/@(fullscreen|unplugged|showdialog|showhint)/gi, '');
-            text = text.trim();
             // remove brackets for hiding step title
-            if (text.match(/^\{([\s\S]+)\}$/))
-                text = text.substr(1, text.length - 2);
+            if (text.match(/\{([\s\S]+)\}/))
```

Comment:
hmm i think maybe something like 
```
if (text.match(/\{([\s\S]+)\}/)?.[1])
     text = text.match(/\{([\s\S]+)\}/)[1].trim()
```
would ensure that we're just getting the text between the brackets.... if someone has a title that's like `{TITLE} other text` it would only translate to `TITLE` but maybe that's okay? would result in a cleaner ID anyway

Reply:
Right, to make it simple, maybe just have the "rules" be that the heading is either all-visible or all-hidden. Heading text that breaks the rules will just need to get fixed when discovered. Go ahead and make the changes as you see fit. Thanks.

Reply:
cool, updated! (also great branch name hahaha)

Reply:
Thanks. BTW, this is the best regex I can come up with that discriminates something out of the first occurrence of "{blahblah123}".

`\{([^\{][\s\S]+?)\}`

yields: "Introduction " when the heading is crazy like:

`jjkkjk }}} {{Introduction }}{{ {more} or {less}`